Work done on API call

filipe-gomes opened the pull request:
1. Imported axios
2. Created empty list for _cachedItems
3. Created getAvailableRooms function to handle api call to populate rooms
4. Created a placeholder for postReservation function, which will be used to write event data back to Astra
5. Created a call for getAvailableRooms in componentDidMount() 
6. Added temporary api call which will be used to populate room list with real data, for testing purposes.

Need to:
1. Resolve CORS errors, so that the api call will work 
2. Update DetailedRoomButton and RoomList files to include additional parameters (roomNumber, roomBuilding) 
3. Test, improve, repeat

filipe-gomes committed:
Added placeholders for API calls

filipe-gomes committed:
Added temporary API call, updated functions, updated manifest with temp url for testing

filipe-gomes committed:
Updated source location to remove temp url

filipe-gomes committed:
Fixed RoomFinder to populate results with items from API call

adastra-rtownsend commented on package.json:
```diff
@@ -24,9 +24,11 @@
     "watch": "webpack --mode development --watch"
   },
   "dependencies": {
+    "@types/jquery": "^3.3.31",
```
do we actualy need jquery or is this residue from something you were trying?

adastra-rtownsend committed:
added a debug manifest

now we can load the product and debug version side by side by pointing to a debug version of the manifest

adastra-rtownsend committed:
fixed bad url

filipe-gomes commented:
Ryan,

That’s residue from something I was trying. I’ll clean it up.

On Mon, Oct 7, 2019 at 8:27 AM Ryan Townsend <notifications@github.com>
wrote:

> *@adastra-rtownsend* commented on this pull request.
> ------------------------------
>
> In package.json
> <https://github.com/adastra-rtownsend/outlook-addin/pull/1#discussion_r332016766>
> :
>
> > @@ -24,9 +24,11 @@
>      "watch": "webpack --mode development --watch"
>    },
>    "dependencies": {
> +    "@types/jquery": "^3.3.31",
>
> do we actualy need jquery or is this residue from something you were
> trying?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/adastra-rtownsend/outlook-addin/pull/1?email_source=notifications&email_token=AKA23PA6CDMI7F3O3TQOGJ3QNM2LJA5CNFSM4IYGLV3KYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCHCQPTQ#pullrequestreview-298125262>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKA23PFHX5UD6UYZ7VSFJYTQNM2LJANCNFSM4IYGLV3A>
> .
>

filipe-gomes committed:
Removed jQuery residue

filipe-gomes committed:
Merge branch 'master' of https://github.com/filipe-gomes/outlook-addin

filipe-gomes committed:
Update to show building name and room number, fixed capacity value

filipe-gomes committed:
Updates

adastra-rtownsend committed:
restored urls for prod

adastra-rtownsend committed:
Merge branch 'master' of https://github.com/filipe-gomes/outlook-addin into pr/1

adastra-rtownsend committed:
reading data from astra-api-bridge

not all data is there yet, e.g. capacity not yet reported

adastra-rtownsend committed:
refining room loading

automatically load rooms from server on component load, show spinner while loading

adastra-rtownsend committed:
picking up actually times from meeting to send to api request

adastra-rtownsend committed:
hacky time change monitoring

checking every 2 seconds to see if start time or end time changed, need to subscribe to office async notification instead, but not yet supported in all platforms.

adastra-rtownsend committed:
added "book room" button

this is where will call the api-bridge to book the room (and event) in Astra Schedule

adastra-rtownsend committed:
post call to reserve room

includes an error bar when it fails

adastra-rtownsend committed:
give more vertical space to the scrollable room list

adastra-rtownsend committed:
cleanup console.log

adastra-rtownsend committed:
UI improvements

fixed background color on room icon, added hover background coloring to items in the room list

adastra-rtownsend committed:
Update RoomList.tsx

adastra-rtownsend committed:
temporarily switching back to example data

cleaned up interface definition for room button (decoupled the source data interface), toggled back to loading sample data -- will add a config setting next to control this,  updated example data interface to match server data

adastra-rtownsend committed:
console.log cleanup

adastra-rtownsend committed:
added settings Dialog

invoked by righ-clicking somewhere in the search area, this from the office fabric react examples, still need to build out settings

adastra-rtownsend committed:
Implementation of Settings Dialog

added default setting values, implemented SettingsDialog

adastra-rtownsend committed:
observing settings

applied logic for usings settings

adastra-rtownsend committed:
minor bugs

fixed capacity for sample data to not be hardcoded

adastra-rtownsend committed:
added npm deploy

deploying to surge.sh for now

adastra-rtownsend committed:
fixed styles

weren't using proper selector to change room icon color.

adastra-rtownsend committed:
only show capacity if it's not empty

only show capacity if it's not empty

adastra-rtownsend committed:
fixed loading and booking logic and spinners

adastra-rtownsend committed:
simulate room booking when using sample data

adastra-rtownsend committed:
refactor of settings

fixed bugs around clearing settings and default values

adastra-rtownsend committed:
Update App.tsx

fixed raced condition around property updates

adastra-rtownsend committed:
added redux

added redux based on this: https://codesandbox.io/s/w02m7jm3q7

adastra-rtownsend committed:
fixed comma positioning

adastra-rtownsend committed:
apply roaming settings for real now

adastra-rtownsend committed:
yet another state sync'ing issue

adastra-rtownsend committed:
demo prep

even on fail, show success message and add room to Outlook location

filipe-gomes committed:
Updated files to match changes made to redux branch in parent fork

filipe-gomes committed:
Merge branch 'rtownsend/real-data' of https://github.com/adastra-rtownsend/outlook-addin

Merging updates from rtownsand/real-data branch from parent fork

filipe-gomes committed:
reverted RoomFinder.tsx back to using real data

filipe-gomes committed:
argargargh

filipe-gomes committed:
Added getCapacity() function

filipe-gomes committed:
Declared capacity in refreshRoomInfo

filipe-gomes committed:
strugling

filipe-gomes committed:
changes
